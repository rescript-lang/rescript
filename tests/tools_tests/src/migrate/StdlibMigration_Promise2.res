let p1 = Js.Promise2.resolve(1)
let p2 = Js.Promise2.reject(Failure("err"))

let all1 = Js.Promise2.all([Js.Promise2.resolve(1), Js.Promise2.resolve(2)])
let all2 = Js.Promise2.all2((Js.Promise2.resolve(1), Js.Promise2.resolve(2)))
let all3 = Js.Promise2.all3((
  Js.Promise2.resolve(1),
  Js.Promise2.resolve(2),
  Js.Promise2.resolve(3),
))

let all4 = Js.Promise2.all4((
  Js.Promise2.resolve(1),
  Js.Promise2.resolve(2),
  Js.Promise2.resolve(3),
  Js.Promise2.resolve(4),
))
let all5 = Js.Promise2.all5((
  Js.Promise2.resolve(1),
  Js.Promise2.resolve(2),
  Js.Promise2.resolve(3),
  Js.Promise2.resolve(4),
  Js.Promise2.resolve(5),
))
let all6 = Js.Promise2.all6((
  Js.Promise2.resolve(1),
  Js.Promise2.resolve(2),
  Js.Promise2.resolve(3),
  Js.Promise2.resolve(4),
  Js.Promise2.resolve(5),
  Js.Promise2.resolve(6),
))

let race1 = Js.Promise2.race([Js.Promise2.resolve(10), Js.Promise2.resolve(20)])

let thenPipe = Js.Promise2.resolve(1)->Js.Promise2.then(x => Js.Promise2.resolve(x + 1))
let thenDirect = Js.Promise2.then(Js.Promise2.resolve(1), x => Js.Promise2.resolve(x + 1))

// Type alias migration
external p2: Js.Promise2.t<int> = "p2"

let catchPipe = Js.Promise2.resolve(1)->Js.Promise2.catch(_e => Js.Promise2.resolve(0))
let catchDirect = Js.Promise2.catch(Js.Promise2.resolve(1), _e => Js.Promise2.resolve(0))
let make1 = Js.Promise2.make((~resolve, ~reject) => resolve(1))

let _ = p2->Js.Promise2.then(x => Js.Promise2.resolve(x + 1))
