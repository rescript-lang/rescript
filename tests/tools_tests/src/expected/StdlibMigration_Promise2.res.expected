let p1 = Promise.resolve(1)
let p2 = Promise.reject(Failure("err"))

let all1 = Promise.all([Promise.resolve(1), Promise.resolve(2)])
let all2 = Promise.all2((Promise.resolve(1), Promise.resolve(2)))
let all3 = Promise.all3((Promise.resolve(1), Promise.resolve(2), Promise.resolve(3)))

let all4 = Promise.all4((
  Promise.resolve(1),
  Promise.resolve(2),
  Promise.resolve(3),
  Promise.resolve(4),
))
let all5 = Promise.all5((
  Promise.resolve(1),
  Promise.resolve(2),
  Promise.resolve(3),
  Promise.resolve(4),
  Promise.resolve(5),
))
let all6 = Promise.all6((
  Promise.resolve(1),
  Promise.resolve(2),
  Promise.resolve(3),
  Promise.resolve(4),
  Promise.resolve(5),
  Promise.resolve(6),
))

let race1 = Promise.race([Promise.resolve(10), Promise.resolve(20)])

let thenPipe = Promise.resolve(1)->Js.Promise2.then(x => Promise.resolve(x + 1))
let thenDirect = Js.Promise2.then(Promise.resolve(1), x => Promise.resolve(x + 1))

// Type alias migration
external p2: promise<int> = "p2"

// let catchPipe = Js.Promise2.resolve(1)->Js.Promise2.catch(_e => Js.Promise2.resolve(0))
// let catchDirect = Js.Promise2.catch(Js.Promise2.resolve(1), _e => Js.Promise2.resolve(0))
let make1 = Promise.make((resolve, reject) => resolve(1))

